service/history/task/task.go

```diff
@@ -320,6 +321,13 @@ func (t *taskBase) HandleErr(
 		return nil
 	}
 
+	if t.attempt > t.maxRetryCount() && common.IsStickyTaskConditionError(err) {
```

Comment:
Please grab a lock when accessing the `attempt` field or call `GetAttempt()` which does it for you. 

The `attempt` field will also be accessed by the split policy to determine the if the task got stuck and split it to a new queue, that's why I added a lock for it.

Reply:
Hmmm. Actually it's probably fine reading that field without a lock in this function as there's won't be any concurrent write to this field (write only happens in the defer func). Thoughts?

Reply:
I think since we have `GetAttempt` it more correct and consistent to just use it (and avoid mislead other reader of related code).
If we want to optimize, we could try RWlock on this later.